sass/cd/cd-header/_cd-language-switcher.scss

```diff
@@ -6,123 +6,203 @@
-    @extend .cd-user-menu__dropdown-link;
+  &.is-active {
+    font-weight: 700;
+  }
+}
 
-    color: var(--cd-white);
+// JS disabled.
+// Language Switcher list.
+.no-js .cd-language-switcher__dropdown {
+  /* Unset the dropdown styles */
+  position: relative;
+  right: unset;
+  left: unset;
+  display: flex;
+  flex-wrap: wrap;
+  padding: 0;
+  background: transparent;
+  box-shadow: none;
+}
 
-    &:hover,
-    &:focus {
-      text-decoration: underline;
-    }
+// Pipe separator.
+.no-js .cd-language-switcher__dropdown li a::before {
+  position: absolute;
+  top: 50%;
+  left: -1px;
+  width: 1px;
+  height: calc(var(--cd-global-header-height) / 3);
+  content: "";
+  transform: translateY(-50%);
+  background: var(--brand-grey);
+
+  [dir="rtl"] & {
+    right: -1px;
+    left: auto;
+  }
+}
 
-    &:focus {
-      outline: 3px solid var(--brand-primary--light);
-    }
+// No pipe on first child.
+.no-js .cd-language-switcher__dropdown li:first-child a::before {
+  background: transparent;
+}
+
+// If there are menus in the Global Header
+// and the Language Switcher is displayed in full,
+// separate with pipes and margins.
+.cd-global-header__language-switcher + .cd-global-header__user-menu {
+  position: relative;
+  // Margin value matches the padding of the other buttons in the Global Header
+  // and takes into account the right padding from the last Language Item.
+  @include desktop {
+    margin-inline-start: 1rem;
   }
 }
 
-.cd-user-menu__dropdown-link {
-  display: block;
-  padding: 12px 0;
-  text-decoration: none;
+// Pipe separator between Language Switcher and following menu.
+.js {
+  .cd-global-header__language-switcher + .cd-global-header__user-menu::before {
+    position: absolute;
+    top: 50%;
+    left: -1px;
+    width: 2px;
+    height: calc(var(--cd-global-header-height) / 3);
+    content: "";
+    transform: translateY(-50%);
+    background: var(--brand-grey);
+  }
 
-  &.is-active {
+  &[dir="rtl"] .cd-global-header__language-switcher + .cd-global-header__user-menu::before {
+    right: -1px;
+    left: auto;
+  }
+}
+
+// Adjust breakpoints for when the Language Switcher displays as a dropdown.
+// This depends on other elements present in the .cd-global-header__actions.
+//
+// Copy the following rules into the sub theme and override breakpoint values.
+@media (min-width: 1024px) {
+
+  /* Language Switcher list. */
+  .cd-language-switcher__dropdown {
+    /* Unset the dropdown styles */
     position: relative;
-    font-weight: 700;
+    right: unset;
+    left: unset;
+    display: flex;
+    flex-wrap: wrap;
+    padding: 0;
+    background: transparent;
+    box-shadow: none;
+  }
 
-    [dir="ltr"] & {
-      padding-right: 12px;
-    }
-
-    [dir="rtl"] & {
-      padding-left: 12px;
-    }
-
-    &::after {
-      position: absolute;
-      right: 0;
-      width: 14px;
-      height: 14px;
-      content: "";
-      background-image: url("../img/icons/selected--highlighted.svg");
```

Comment:
This is the checkmark icon that was beside the active language in the dropdown. The SVG can also be removed.